reflectapi-schema/src/lib.rs

```diff
@@ -473,10 +478,30 @@ impl TypeReference {
 
     fn rename_type(&mut self, search_string: &str, replacer: &str) {
         self.name = rename_type_or_module(&self.name, search_string, replacer);
-        for param in self.parameters.iter_mut() {
+        for param in self.arguments.iter_mut() {
             param.rename_type(search_string, replacer);
         }
     }
+
+    fn instantiate(self, subst: &Subst) -> TypeReference {
```

Comment:
where is it used? what are you trying to do here? This seems like belongs to codegen... It is also very confusing that TypeReference can be not instanciated. It is always instanciated, TypeParameter is not....

Reply:
It's used in code like the following. There can implicitly be generics in `TypeReference`s which this function tries to replace with types. `TypeReference::instantiate` is called indirectly by `Struct::instantiate` and `Enum::instantiate` which call `Field::instantiate`. 

I do think that the representation of types is not ideal, in particular there is no way to distinguish between a type named `T` and `T` representing a reference to a type parameter. This was the most minimal code I could come up with that mostly works.

```rust
            crate::Type::Struct(strukt) => match self.struct_to_schema(
                schema,
                kind,
                &strukt.clone().instantiate(&ty_ref.arguments),
            )
```

Reply:
and where the Struct::instantiate is used?
in general you can not implement this sort of functionality without having access to the type definition.

Reply:
I think I understand what you are doing here. I think I do something similar in the TS and Rust codegen. Check resolve type ref functions.

Reply:
The above code sample is verbatim from `codegen/openapi.rs`. Given a type reference with args `S<int, bool>` referencing generic struct `S<T, U> { t: T, u: U }`. Instantiate takes a generic struct with `k` type parameters and `k` type arguments and returns a non-generic struct with the parameters filled in.

`instantiate(S<T, U> { t: T, u: U}, [int, bool]) returns non generic struct S { t: int, u: bool }`.

Reply:
Without the instantiation, the openapi spec will incorrect state that `S.t` has type `T` instead of `int` or whatever.

Reply:
I think this code belongs to codegen... Maybe some helper wrap struct is needed, not sure. It feels very foreign to have it in the schema.

Reply:
Ok agree that I didn't like `instantiate` being a public function. I'll move the instantiation functions as private free functions.